src/pages/Summary.jsx

```diff
@@ -165,6 +168,63 @@ const Summary = () => {
         };
     }, [user]);
 
+    // Upload unsynced data from local database
+    useEffect(() => {
+        if (!user) return;
+        const uploadUnsyncedData = async () => {
+            const unsyncedLogs = await ldb.logs.where("unsynced").equals(1).toArray();
+            if (unsyncedLogs.length === 0) return;
+
+            const keys = checkKeys();
+            if (keys === "discreet") return;
+
+            for (let log of unsyncedLogs) {
+                let data = new FormData();
+                data.append("timezone", DateTime.local().zoneName);
+                data.append("mood", log.mood);
+                data.append("journal", log.journal);
+                data.append("average", log.average);
+                data.append("keys", JSON.stringify(checkKeys()));
+                data.append("addFlag", log.addFlag ? log.addFlag + " offlineSync:" + log.timestamp : " offlineSync:" + log.timestamp);
+                data.append("song", log.song?.uri || "");
+                if (log.files && log.files.length > 0) {
+                    for (let file of log.files) {
+                        data.append("file", file);
+                    }
+                }
+                data.append("audio", log.audio);
+
+                let response;
+                try {
+                    response = await fetch(`${BASE_URL}/moodLog`, {
+                        method: "POST",
+                        headers: {
+                            "Authorization": `Bearer ${await getIdToken(user)}`,
+                        },
+                        body: data
+                    });
+                    // delete the unsynced log from the local database
+                    await ldb.logs.where("timestamp").equals(log.timestamp).delete();
```

Comment:
this deletion should happen after the response.ok check

Reply:
done